pinot-controller/src/main/java/com/linkedin/pinot/controller/api/restlet/resources/PinotSegmentRestletResource.java

```diff
@@ -230,19 +232,38 @@ protected Representation toggleSegmentState(String tableName, String segmentName
 
     JSONArray ret = new JSONArray();
     String tableNameWithType = null;
-    if ((tableType == null || TableType.OFFLINE.name().equalsIgnoreCase(tableType))
+    List<String> segmentsToToggle = new ArrayList<>();
+    String offlineTableName = TableNameBuilder.OFFLINE_TABLE_NAME_BUILDER.forTable(tableName);
+    String realtimeTableName = TableNameBuilder.REALTIME_TABLE_NAME_BUILDER.forTable(tableName);
+
+    if (segmentName != null) {
+      segmentsToToggle = Collections.singletonList(segmentName);
+    } else {
+      // Check if there is a realtime table. If there is, remove all segments for realtime
+      if (_pinotHelixResourceManager.hasRealtimeTable(tableName)) {
+        segmentsToToggle.addAll(_pinotHelixResourceManager.getAllSegmentsForResource(realtimeTableName));
+      }
+      segmentsToToggle.addAll(_pinotHelixResourceManager.getAllSegmentsForResource(offlineTableName));
+    }
+    if ((tableType == null
```

Comment:
If table type is null, then do you want to toggle all segments? Are we  toggling only offline segments in that case?